internal/pkg/cli/pipeline_init.go

```diff
@@ -192,6 +199,57 @@ func (o *initPipelineOpts) Ask() error {
 	return nil
 }
 
+// validateDuplicatePipeline checks that the pipeline name isn't already used
+// by another pipeline, whether it's been deployed or just has a local manifest file.
+// We check for the existance of the name and the namespaced name to reduce
+// potential confusion with a legacy pipeline.
+func (o *initPipelineOpts) validateDuplicatePipeline() error {
+	// make sure pipeline isn't already deployed
+	names, err := o.codePipeline.ListPipelineNamesByTags(map[string]string{
+		deploy.AppTagKey: o.appName,
+	})
+	if err != nil {
+		return fmt.Errorf("validate if pipeline exists: %w", err)
+	}
+
+	fullName := fmt.Sprintf(fmtPipelineName, o.appName, o.name)
+	for _, name := range names {
+		if strings.EqualFold(name, o.name) || strings.EqualFold(name, fullName) {
```

Comment:
It'd be helpful for future readers if we add a comment here explaining that we are validating against both the new namespaced pipeline name (the `fullname`) and the old non-namespaced name.

Reply:
Explanation is in the docs for the function :blush: